core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -777,6 +777,12 @@ class GroupCoordinator(val brokerId: Int,
 
     maybePrepareRebalance(group, s"Adding new member $memberId")
     group.removePendingMember(memberId)
```

Comment:
For logical consistency, I'd suggest this line go above `maybePrepareRebalance`. Removal of the pending member may resolve the rebalance.